client/landing/gutenboarding/components/domain-picker/index.tsx

```diff
@@ -93,87 +104,95 @@ const DomainPicker: FunctionComponent< Props > = ( {
 		[ search, queryParameters ]
 	);
 
-	let numberOfFreeDomains = 0;
-	let numberOfPaidDomains = 0;
-	const suggestions = allSuggestions?.filter( suggestion => {
-		if (
-			suggestion.is_free &&
-			! numberOfFreeDomains &&
-			suggestion.domain_name !== currentDomain?.domain_name
-		) {
-			numberOfFreeDomains++;
-			return suggestion;
-		}
-		if ( ! suggestion.is_free && numberOfPaidDomains < 5 ) {
-			numberOfPaidDomains++;
-			return suggestion;
-		}
-	} );
-
-	const handleDomainClick = ( suggestion: DomainSuggestion ) => {
-		if ( suggestion.is_free ) {
-			onDomainSelect( suggestion );
-		} else {
-			onDomainPurchase( suggestion );
-		}
-	};
+	const freeSuggestions = allSuggestions?.filter( suggestion => suggestion.is_free );
+	const paidSuggestions = allSuggestions?.filter( suggestion => ! suggestion.is_free );
+
+	// Recommend paid domain with exact site title match with highest relevance score.
+	const recommendedSuggestion = paidSuggestions?.find( suggestion =>
+		suggestion.match_reasons.includes( 'exact-match' )
+	);
 
 	return (
 		<Panel className="domain-picker">
 			<PanelBody>
 				<PanelRow className="domain-picker__panel-row">
-					<div className="domain-picker__choose-domain-header">
-						{ NO__( 'Choose a new domain' ) }
+					<div className="domain-picker__header">
+						<div className="domain-picker__header-title">{ NO__( 'Choose a new domain' ) }</div>
+						<Button className="domain-picker__close-button" isTertiary onClick={ () => onClose() }>
+							{ NO__( 'Skip for now' ) }
+						</Button>
+					</div>
+					<div className="domain-picker__search">
+						<SearchCard
+							className="domain-picker__search-field"
+							inputLabel={ label }
+							placeholder={ label }
+							onSearch={ setDomainSearch }
+							value={ domainSearch }
+							compact
+							hideClose
+							disableAutocorrect
+						/>
+						<div className="domain-picker__connect-domain">
+							<span>{ NO__( 'Already have a domain?' ) }</span>&nbsp;
```

Comment:
**FYI (no need to make this change, your call)**, I'm not sure what the motivations were here. I know I've used `&nbsp;` liberally recently. React has some funny spacing rules. If you want a space, you can add a JavaScript literal space:

```suggestion
							<span>{ NO__( 'Already have a domain?' ) }</span>{ ' ' }
```